src/Features/Core/Portable/ConvertIfToSwitch/AbstractConvertIfToSwitchCodeRefactoringProvider.cs

```diff
@@ -1,256 +1,119 @@
-                    currentStatement = _syntaxFacts.GetNextExecutableStatement(currentStatement);
-                }
-                while (currentStatement != null);
-            }
+        public sealed class ConstantPattern : Pattern
+        {
+            public readonly SyntaxNode ExpressionSyntax;
 
-            private Task<Document> UpdateDocumentAsync(
-                SyntaxNode root,
-                Document document,
-                TIfStatementSyntax ifStatement,
-                IEnumerable<(IEnumerable<IPattern<TSwitchLabelSyntax>> patterns, TStatementSyntax statement)> sections)
+            public ConstantPattern(IOperation operation)
             {
-                var generator = SyntaxGenerator.GetGenerator(document);
-                var sectionList =
-                    sections.Select(s => generator.SwitchSectionFromLabels(
-                        labels: s.patterns.Select(p => p.CreateSwitchLabel()),
-                        statements: GetSwitchSectionBody(s.statement))).ToList();
-
-                if (_switchDefaultBodyOpt.HasValue)
-                {
-                    sectionList.Add(generator.DefaultSwitchSection(GetSwitchSectionBody(_switchDefaultBodyOpt.Value)));
-                }
-
-                var ifSpan = ifStatement.Span;
-                var @switch = CreateSwitchStatement(ifStatement, _switchExpression, sectionList);
-                var nodesToRemove = GetSubsequentStatements(ifStatement)
-                    .Skip(1).Take(_numberOfSubsequentIfStatementsToRemove).ToList();
-                root = root.RemoveNodes(nodesToRemove, SyntaxRemoveOptions.KeepNoTrivia);
-
-                var lastNode = nodesToRemove.LastOrDefault() ?? ifStatement;
-                @switch = @switch.WithLeadingTrivia(ifStatement.GetLeadingTrivia())
-                                 .WithTrailingTrivia(lastNode.GetTrailingTrivia())
-                                 .WithAdditionalAnnotations(Formatter.Annotation);
-
-                root = root.ReplaceNode(root.FindNode(ifSpan), @switch);
-                return Task.FromResult(document.WithSyntaxRoot(root));
+                ExpressionSyntax = operation.Syntax;
             }
+        }
 
-            protected abstract SyntaxNode CreateSwitchStatement(TIfStatementSyntax ifStatement, TExpressionSyntax expression, List<SyntaxNode> sectionList);
-
-            protected abstract TExpressionSyntax UnwrapCast(TExpressionSyntax expression);
-
-            protected abstract bool EndPointIsReachable(TIfStatementSyntax ifStatement);
-
-            protected abstract bool CanConvertIfToSwitch(TIfStatementSyntax ifStatement);
-
-            protected abstract IEnumerable<TExpressionSyntax> GetLogicalOrOperands(TExpressionSyntax syntaxNode);
-
-            protected abstract IEnumerable<(TExpressionSyntax, TStatementSyntax)> GetIfElseStatementChain(TIfStatementSyntax ifStatement);
+        public sealed class RelationalPattern : Pattern
```

Comment:
consider moving all the patterns to another file.  i think it would also make the diff easier to read.